docs/providers/tutorial-create-a-provider.md

```diff
@@ -0,0 +1,754 @@
+
+When things don't work as expected, you can ask for help or guidance online[^serverlesswasm].
+
+[^serverlesswasm]: The space of serverless WebAssembly is still
+    young and a lot of package maintainers can make the change or help you make the change to get a library to compile and
+    run for WebAssembly. Sometimes it means finding out support is [already there](https://github.com/gyscos/zstd-rs/pull/139),
+    sometimes it means [adding items to their roadmap](https://github.com/awslabs/aws-sdk-rust/discussions/679)
+
+#### Checkpoint
+
+At that point you must know
+- what configuration values are necessary to communicate with the third party
+  resource
+- a documentation/reference to be able to programmatically communicate with said
+  resources
+
+### Clone the Providers repo
+
+The current Fiberplane [providers repo](https://github.com/fiberplane/providers)
+contains a "sample"
+[provider](https://github.com/fiberplane/providers/tree/main/providers/sample)
+that is a good base to make your provider from: our CI pipeline guarantees that
+the sample provider is always up-to-date with our supporting libraries, so it is
+the quickest way to obtain a starting template.
+
+Clone the repository and keep the location of the sample provider somewhere handy:
+```console
+$ git clone https://github.com/fiberplane/providers
+$ export SAMPLE_PROVIDER_PATH="$(pwd)/providers/providers/sample"
+```
+
+#### Checkpoint
+
+If you never restart your terminal session again, we will have a valid path for
+`${SAMPLE_PROVIDER_PATH}`. We will not use it for long, but the point is that
+you should have a fresh copy of the Fiberplane/providers repository.
+
+
+
+## Instantiate a Daemon and side-load your sample provider for use in Studio
+
+### Create your new provider repo, and test sample compilation
+
+It's time to create your provider repo. To avoid the usual generic names, we
+will create a `catnip` provider.
+
+```console
+$ mkdir catnip_provider
+$ cd catnip_provider
+$ cp -r "${SAMPLE_PROVIDER_PATH}"/* .
+$ git init .
+```
+
+Once the repo is created, use the Cargo configuration feature to make the
+compiler default to Web Assembly compilation. Create a `.cargo` folder at the
+root of your repository, and a `config.toml` file[^profilerelease] inside:
+
+```toml
+# In .cargo/config.toml
+[build]
+target = "wasm32-unknown-unknown"
+```
+
+[^profilerelease]: You can also add a few other flags if you want. At Fiberplane
+    we add a few flags for building the providers using the least disk-space
+    possible, you can see them in our
+    [repository](https://github.com/fiberplane/providers/blob/main/providers/.cargo/config)
+    
+The last step to setup your replica of the sample provider is to own it by
+editing the `cargo.toml` file to set the name of the library:
+
+```toml
+# In Cargo.toml
+name = "catnip_provider"
+# ... Also edit the other metadata fields as you see fit, no `workspace` related metadata must stay in that section
```

Comment:
sample values for the other metadata could be nice, but maybe that was only a hiccup for me due to lack of familiarity with rust

Reply:
I'll add a link to the example repo